### ISSUE TITLE
Forge eats all exceptions

### ISSUE THREAD

**Issue by calclavia** [NOVA-Team/NOVA-Core#168]

**Comment by Caellian** [comment]
It must be really hungry.
Is there a way around this without modifying forge?
Why does NOVA depend on forge? Can't it just support forge and be built on
MCP.
That would make way more sense considering wrappers should be built on
"original" code instead of deobfuscated code modified by people not
officialy supported or affiliated with Mojang.
I do get that it's probably a pain to write everything from 0 and that this
wasn't a point of NOVA to start with, but it makes this code more stable
and you will not have to deal with bugs like these.

**Comment by gjgfuj** [comment]
1: forge is way more stable than vanilla.
2: it's easier for the user to install, if they can just use forge.
3: If a mod requires a native plugin they definitely won't want to build
against non-forge.
4: It's much better, because you can use FML and stuff, plus we can always
build around everything.

**Comment by Caellian** [comment]
I guess what I was trying to say is that if you had control over forge too,
it would make more sense for forge to extend NOVA instead the other way
around.
I agree with some of your arguments however.

**Comment by gjgfuj** [comment]
No, NOVA builds on top of forge, because forge is a low-level abstraction
that builds on top of minecraft, but it doesn't abstract much of MC and
you're writing a mod with actual MC code. While NOVA is a complete
abstraction, that /could/ be built on top of pure minecraft but isn't
because it's much easier for us, much easier for the users, and much easier
for the modders.

**Comment by AEnterprise** [comment]
maybe talk to lex about this? maybe a special type of exception that nova can then extend to get arround this?

**Comment by RX14** [comment]
Why would forge eat the exceptions anyways?

**Comment by AEnterprise** [comment]
probably cause there are mods that are throwing exceptions internaly and in some cases they are not handled properly (i know AE2 uses alot of internal exceptions, duno a mod that handles the badly directly but it's the only thing that would make sense)

**Comment by gjgfuj** [comment]
If they're not handled properly.... They should crash.

**Comment by RX14** [comment]
@gjgfuj Exactly, this just makes the situation worse for every other modder that **actually wants to debug their mod**

**Comment by gjgfuj** [comment]
I have the same opinion about avoiding npes. But yknow.

**Comment by RX14** [comment]
What do you mean about avoiding NPEs?

**Comment by gjgfuj** [comment]
Well in my opinion optional is a completely pointless bit of extra rubbish.
But yknow, people can think whatever they want.

**Comment by RX14** [comment]
How do optionals make the situation worse for people who want to debug their mod?

**Comment by Kubuxu** [comment]
Forge eats some exceptions. Worst thing is that it fails to log them and errors out.

Optional is good. It forces you to think about performing a null check. Many, many bugs in forge mods are NPEs, many bugs in Java programs are NPEs.

**Comment by RX14** [comment]
If forge even just logged the exception it would be great, but no. Has anyone ACTUALLY asked in #MinecraftForge about this?

**Comment by Victorious3** [comment]
Working with MCP and still being compatible with other mods is *impossible* to *really hard and not worth it*, @Caellian 

Also, most of those exceptions are eaten by google's event bus, and that's nothing that Forge can change.

**Comment by RX14** [comment]
Why is the eventbus eating exceptions, and if we can't change it, we should surround our event handlers with try-catch to print the exceptions.

**Comment by SoniEx2** [comment]
Throw `Error` Kappa
